cpp/src_prims/metrics/adjustedRandIndex.h

```diff
@@ -47,6 +50,58 @@ struct nCTwo {
   }
 };
 
+template <typename DataT, typename IdxT>
+struct Binner {
+  Binner(DataT minL) : minLabel(minL) {}
+
+  DI int operator()(DataT val, IdxT row, IdxT col) {
+    return int(val - minLabel);
+  }
+
+ private:
+  DataT minLabel;
+};  // struct Binner
+
+/**
+ * @brief Function to count the number of unique elements in the input array
```

Comment:
Would be nice to see this pulled out into its own prim eventually as I think it can be generally useful. The `labels.h` prims originally used to be nested in a folder named `array` for this reason.